PProPanel/ext.js

```diff
@@ -222,11 +222,11 @@ function loadJSX() {
 	var extensionPath = csInterface.getSystemPath(SystemPath.EXTENSION);
 
 	// load general JSX script independent of appName
-	const extensionRootGeneral = extensionPath + '/jsx/';
+	var extensionRootGeneral = extensionPath + '/jsx/';
```

Comment:
>Because const is an [ES6 standard](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/const), and I don't think it's supported by the CEF version used in CEP.

Ahh. This probably depends a bit on what the minimum supported CEP version should be for the PProPanel sample. @bbb999: how far back in the release history is the sample intended to support?

@mBeierl You can easily find this out in [the docs](https://github.com/Adobe-CEP/CEP-Resources), but CEP 5.2 is equivalent to **Chrome 27**, while CEP 6.1+ is equivalent to **Chrome 41**. [This](http://caniuse.com/#compare=chrome+27,chrome+41) is a quick view comparing what does and doesn't work in those browsers. If you check [`const` support specifically](http://caniuse.com/#feat=const) you can see that there are some extra notes:

* **Chrome 27** - `const` is recognized only when _not_ in `strict` mode. No block scoping.
* **Chrome 41** - `const` has block scoping in `strict` mode, but no block scoping otherwise.

As `strict` mode is _not_ used in the scripts in question, `const` would be supported identically - it works, just not in a block-scoped fashion. **_Personally_**, I would suggest making use of it where possible as it is a good enhancement and improves readability. I'd ask @bbb999 to weigh in on this, though.

>In my projects I use babel to transpile my source code, that's why I used const there in the first place.

Have you checked out the [TypeScript sample](https://github.com/Adobe-CEP/Samples/tree/master/TypeScript) yet? The environment there is specifically set up to provide you with autocompletion and IntelliSense support for ExtendScript files (via [type declaration files](https://github.com/Adobe-CEP/Samples/tree/master/TypeScript/PProPanel-vscode/dom_app/types). It's also pre-configured (see the [`dom_html/tsconfig.json` file](https://github.com/Adobe-CEP/Samples/blob/master/TypeScript/PProPanel-vscode/dom_html/tsconfig.json)) to transpile _both_ TypeScript and JavaScript (`--allowJs` into panel-compatible JavaScript (es5 target, with a note to switch to es3 if issues arise).